src/traits.rs

```diff
@@ -191,3 +191,42 @@ pub trait AggregateAuthenticator:
         messages: &[&[u8]],
     ) -> Result<(), Error>;
 }
+
+/// Trait impl'd by encryption keys in symmetric cryptography
+///
+pub trait EncryptionKey:
+    ToFromBytes + 'static + Serialize + DeserializeOwned + Send + Sync
+{
+    fn generate<R: CryptoRng + RngCore>(rng: &mut R) -> Self;
+}
+
+/// Trait impl'd by symmetric ciphers.
+///
+pub trait Cipher {
+    /// Encrypt `plaintext` and write result to `buffer` using the given IV
+    fn encrypt(&self, iv: &[u8], plaintext: &[u8], buffer: &mut [u8]) -> Result<(), Error>;
```

Comment:
Wondering if we should hide this from the end-user `buffer: &mut [u8])`, by experiencer it's very common to make mistakes here or be inconvenient, ie by providing wrong size input buffers, even the fact to be prepared to have an existing buffer already. Wdyt?

Reply:
That makes sense. I just copied the interface from rustcrypto, assuming that was the common way to do it in rust. I think we should return a new array with the result then.